### PR TITLE
#164049106(v2) Fix booked rooms GraphQL endpoints

### PR DESCRIPTION
**Pull request by dnuwa**
#### Description
Fixes the get most-booked and least-booked rooms GraphQL endpoints
GraphQL endpoints for most-booked and least-booked rooms return data different from same RESTful endpoints. This PR Modifies the GraphQL endpoints to return data similar to the RESTful endpoints.
Also enables a user to set a limit of the number of results they would like to see. 
Creates a single GraphQL endpoint that can be used to return both the most booked and least booked rooms analytics by specifying the `limit` and `criteria` or not to return analytics for all.
#### Type of change
chore (Fix get most-booked and least-booked rooms GraphQL endpoints)
#### How Has This Been Tested?
- git pull the branch `ch-most-and-least-booked-rooms-endpoints-v2-164049106`
- run the queries below.

`analytics for most booked rooms`
````
{
 analyticsForBookedRooms(startDate: "Feb 14 2019", endDate: "Feb 14 2019"){
     analytics{
      roomName
      count
      meetings
      percentage
    }
    }
}
````
`analytics for MostBookedRooms with a limit` 
````
{
 analyticsForBookedRooms(startDate: "Feb 14 2019", endDate: "Feb 14 2019", limit:3, criteria:"most_booked"){
     analytics{
      roomName
      count
      meetings
      percentage
      roomName
    }
    }
}
````
`analytics for LeastBookedRooms with a limit`
````
{
 analyticsForBookedRooms(startDate: "Feb 14 2019", endDate: "Feb 14 2019", limit:10, criteria:"least_booked"){
     analytics{
      roomName
      count
      meetings
      percentage
      roomName
    }
    }
}
````

#### Checklist:
 - [x] My code follows the style guidelines of this project
 - [x] I have linted my code prior to submission
 - [x] I have made the corresponding changes to the documentation
 - [x] My changes generate no new warnings
 - [x] I have added tests that prove my feature works
 - [x] New and existing unit tests pass locally with my changes
#### What are the relevant pivotal tracker stories?
[#164049106](https://www.pivotaltracker.com/story/show/164049106)

### CODE REVIEW

File: api/room/schema_query.py [hunk 1]
```diff
@@ -18,6 +19,37 @@
...
+def resolve_booked_rooms_analytics(*args):
+    instance, info, start_date, end_date, criteria, limit = args
+    query = Room.get_query(info)
+    location_id = admin_roles.user_location_for_analytics_view()
+    active_rooms = query.filter(
+        RoomModel.state == "active",
+        RoomModel.location_id == location_id)
+    booked_rooms = get_most_and_least_booked_rooms(
+        instance, active_rooms, start_date, end_date, limit, criteria)
+    rooms_booked_per_week = Analytics(
```
Comment (Bonifase) [inline api/room/schema_query.py]: Why` rooms_booked_per_week`?

Comment (dnuwa) [reply api/room/schema_query.py]: @Bonifase This was a variable used in the previous implementation but looking at it now, it doesn't apply here. That's why I've had to rename it as `booked_rooms_statistics`.

File: api/room/schema_query.py [hunk 2]
```diff
@@ -18,6 +19,37 @@
...
+    booked_rooms_statistics = Analytics(
+        analytics=booked_rooms
+    )
+    return booked_rooms_statistics
+
+
+def get_most_and_least_booked_rooms(*args):
+    instance, active_rooms, start_date, end_date, criteria, limit = args
+    if limit and criteria == "least_booked":
+        limit = (-limit)
```
Comment (Bonifase) [inline api/room/schema_query.py]: Is this assignment necessary here, can we just pass `-limit` directly to the function below?

Comment (dnuwa) [reply api/room/schema_query.py]: passing `-limit` directly raised an error because of the `-` that why it was assigned before being used.

**Comment by dnuwa** [comment]
> Fix the merge conflict

@Bonifase this has been worked on.

**Pull request merged**